napari/_qt/qt_viewer.py

```diff
@@ -905,6 +905,7 @@ def toggle_console_visibility(self, event=None):
 
         if viz:
             self.dockConsole.raise_()
+            self.dockConsole.setFocus()
```

Comment:
Why this is required?

Reply:
Do you mean `raise_()` should set the focus without this? In my testing this did not seem to work.

Another option would instead be `self.console.setFocus()` but this seemed more intuitive to me as the rest of the function is operating on the `dockConsole` rather than the `console` directly. I'm open to the change though; it would remove the need for the change in  
`napari/_qt/widgets/qt_viewer_dock_widget.py`.

Reply:
you already call setFocus in `napari-console` code, so I do not understand why you need to call setFocus here.

Reply:
Ah I see, actually in `napari-console` it only calls `self.setFocusProxy(self._control)` but never actually calls `setFocus` (except in the test).